test/suites/dev/test-storage-growth/test-evm-store-storage-growth.ts

```diff
@@ -0,0 +1,78 @@
+import "@moonbeam-network/api-augment";
+import {
+  TransactionTypes,
+  beforeAll,
+  deployCreateCompiledContract,
+  describeSuite,
+  expect,
+} from "@moonwall/cli";
+import { createEthersTransaction, sendRawTransaction } from "@moonwall/util";
+import { expectEVMResult } from "helpers/eth-transactions";
+import { expectOk } from "helpers/expect";
+import { Abi, encodeFunctionData } from "viem";
+
+describeSuite({
+  id: "D4002",
+  title: "Storage growth limit - New Entries",
+  foundationMethods: "dev",
+  testCases: ({ context, it, log }) => {
+    let storageLoopAddress: `0x${string}`;
+    let storageLoopAbi: Abi;
+    // Number of bytes added to storage for a new entry.
+    const ACCOUNT_STORAGE_SIZE = 116;
+    // Ratio of gas to storage growth. (BlockGasLimit (15_000_000) / BlockStorageLimit (40kb))
+    const GAS_LIMIT_STORAGE_GROWTH_RATIO = 366;
+    beforeAll(async () => {
+      const { contractAddress, abi } = await deployCreateCompiledContract(context, "StorageLoop");
+      storageLoopAddress = contractAddress;
+      storageLoopAbi = abi;
+
+      await context.createBlock();
+    });
+
+    for (const txnType of TransactionTypes) {
+      it({
+        id: `T0${TransactionTypes.indexOf(txnType) + 1}`,
+        title: "should out of gas when gas provided is not enough to cover storage growth",
+        test: async function () {
+          // Tx is creating 5 new storage entries. So, required gas is:
+          // (5 * ACCOUNT_STORAGE_SIZE) * GAS_LIMIT_STORAGE_GROWTH_RATIO = 212_280
+          // Execute tx with insufficient gas limit
+          const rawSigned = await createEthersTransaction(context, {
+            to: storageLoopAddress,
+            data: encodeFunctionData({
+              abi: storageLoopAbi,
+              functionName: "store",
+              // for each transaction type, we add 5 new storage entries
+              args: [5 + 5 * TransactionTypes.indexOf(txnType)],
+            }),
+            gasLimit: 212_270,
+          });
+
+          const { result } = await context.createBlock(rawSigned);
+          // Check that the transaction failed with an out of gas error
+          expectEVMResult(result!.events, "Error", "OutOfGas");
+        },
+      });
+
+      it({
+        id: `T0${TransactionTypes.indexOf(txnType) + 4}`,
+        title: "should successfully execute when updating existing storage entries (no growth)",
+        test: async function () {
+          // Update 5 existing storage entries. So, required gas should be less than 212_280
+          const rawSigned = await createEthersTransaction(context, {
+            to: storageLoopAddress,
+            data: encodeFunctionData({
+              abi: storageLoopAbi,
+              functionName: "store",
+              args: [5],
+            }),
+            gasLimit: 50_000,
+          });
+
+          await expectOk(context.createBlock(rawSigned));
```

Comment:
There is no need to use `expectOk` here, instead you can use the `{ allowFailures: false }` option inside `createBlock` function:
```suggestion
          await context.createBlock(rawSigned, { allowFailures: false });
```

Reply:
We can leave the expectOk to indicate that is the "testing" part of the test, to verify the transaction was ok. Not a big deal